src/confd/yang/infix-routing@2023-11-23.yang

```diff
@@ -106,7 +131,29 @@ module infix-routing {
     }
 
     /* OSPF */
-    /*
+
+    augment "/ietf-r:routing/ietf-r:control-plane-protocols/ietf-r:control-plane-protocol/ospf:ospf" {
+      description "ietf-ospf does not contain redistribution
+	7. How should route redistribution be configured? I see ietf-rip.yang has a separate
+        container for that purpose, but ietf-ospf.yang (and other IGP modules) don't do the
+        same. I also noticed the BGP model is using definition from ietf-routing-policy.yang.
+
+        Different vendors handle redistribution in different ways. This could be added with
+        an augmentation if there were agreement.
+
+        https://marc.info/?l=ms-ospf&m=166258444409552&w=2";
+	container redistributes {
+	  list redistribute {
+	    key "redistribute";
+	    description "Redistribute protocols into OSPF";
+	    leaf redistribute {
+	      type infix-distribute-protocol;
+	      description "Set protocol to redistribute";
```

Comment:
This feels .. odd.  How about something like:

```yang
	container redistributed {
	  list redistribute {
	    key "redistribute";
	    description "Redistribute protocols into OSPF";
	    leaf proto {
	      type infix-distribute-protocol;
	      description "Set protocol to redistribute";
```

what do you think, @wkz ?

Reply:
Key must be proto in your example

Reply:
Yeah, sorry about that.

Reply:
I fell it should be:
	container redistribute {
	  list redistribute {
	    key "proto";
	    description "Redistribute protocols into OSPF";
	    leaf proto {
	      type infix-distribute-protocol;
	      description "Set protocol to redistribute";
	      
then you can do:
admin@dut3:/config/routing/control-plane-protocol/ietf-ospf:ospfv2/name/default/> no ospf redistribute
in the CLI, which fells logical to me.

Reply:
Another solution is to name the container with an ending [d|s] but then the CLI will not autocomplete it when want to remove all redistribute, but that feels like a secondary issue that should be fixed inte CLI.

Maybe container redistribute[s|d] is the best approach.